tests/e2e/vmservice_utils.go

```diff
@@ -714,93 +719,92 @@ func waitNverifyPvcsAreAttachedToVmsvcVm(ctx context.Context, vmopC ctlrclient.C
+		}
 
-	// Extract the device name
-	parts := strings.Split(strings.TrimSpace(diskCheckResults[0].Stdout), "/")
-	if len(parts) < 7 {
-		framework.Failf("Unexpected ls output: %s", diskCheckResults[0].Stdout)
+		// Skip root disk
+		if dev == "/dev/sda" {
+			fmt.Printf("Resolved device is root disk (/dev/sda). Skipping PVC format for UUID %s on VM %s\n", diskUuid, vmIp)
+
+			// 🔎 Debug check: list all disks inside the VM
+			checkResults := execSshOnVmThroughGatewayVm(vmIp, []string{
+				"lsblk -o NAME,SIZE,TYPE,MOUNTPOINT",
+				"sudo fdisk -l || true",
+				"ls -l /dev/disk/by-id/ | grep wwn || true",
+			})
+			fmt.Println("Disk check results:")
+			for _, r := range checkResults {
+				fmt.Println(r.Stdout)
+			}
+
+			return "" // return safely instead of failing
+		}
+
+		break
 	}
-	dev := "/dev/" + parts[6]
-	fmt.Println("Device:", dev)
 
-	gomega.Expect(dev).ShouldNot(gomega.Equal("/dev/"))
-	framework.Logf("Found device %s for disk with UUID %s", dev, diskUuid)
+	if dev == "" || dev == "/dev/" {
+		framework.Failf("Failed to resolve valid device path for UUID %s on VM %s (got %s)", p, vmIp, dev)
+	}
 
+	fmt.Println("Resolved device:", dev)
 	partitionDev := dev + "1"
-	fmt.Println("Partition Device:", partitionDev)
 
-	// Unmount any existing partitions on the device
-	unmountCommands := []string{
-		fmt.Sprintf("sudo umount %s* || true", dev),
-	}
-	res := execSshOnVmThroughGatewayVm(vmIp, unmountCommands)
-	fmt.Println("Unmount Results:", res)
+	// Unmount if already mounted
+	execSshOnVmThroughGatewayVm(vmIp, []string{
+		fmt.Sprintf("sudo umount -f %s || true", partitionDev),
+		fmt.Sprintf("sudo umount -f %s || true", dev),
+	})
 
-	// Partition and format the disk
+	// Partition fresh (wipe GPT headers, then create ext4)
 	partitionCommands := []string{
+		fmt.Sprintf("sudo wipefs -a %s || true", dev),
 		fmt.Sprintf("sudo parted --script %s mklabel gpt", dev),
-		fmt.Sprintf("sudo parted --script -a optimal %s mkpart primary 0%% 100%%", dev),
-		"lsblk -l",
-		fmt.Sprintf("sudo mkfs.ext4 %s", partitionDev),
+		fmt.Sprintf("sudo parted --script -a optimal %s mkpart primary ext4 0%% 100%%", dev),
+		"sleep 2",
+		fmt.Sprintf("sudo mkfs.ext4 -F %s", partitionDev),
 	}
-	res = execSshOnVmThroughGatewayVm(vmIp, partitionCommands)
-	fmt.Println("Partitioning Results:", res)
+	execSshOnVmThroughGatewayVm(vmIp, partitionCommands)
 
-	// Mount the new partition
+	// Mount point
 	volMountPath := "/mnt/volume" + strconv.Itoa(mountIndex)
 	volFolder := volMountPath + "/data"
 	mountCommands := []string{
 		fmt.Sprintf("sudo mkdir -p %s", volMountPath),
-		fmt.Sprintf("sudo mount %s %s", partitionDev, volMountPath),
+		fmt.Sprintf("sudo mount -o discard %s %s", partitionDev, volMountPath),
 		fmt.Sprintf("sudo mkdir -p %s", volFolder),
 		fmt.Sprintf("sudo chmod -R 777 %s", volFolder),
-		fmt.Sprintf("bash -c 'df -Th %s | tee %s/fstype'", partitionDev, volFolder),
+		fmt.Sprintf("df -Th %s | tee %s/fstype", partitionDev, volFolder),
```

Comment:
Your this new change should not break existing functionality. Please run complete vm service regression suite and attach results here

Reply:
pre-checkin has passed , there also it does this check